public/js/draw/setup.js

```diff
@@ -86,19 +86,47 @@ $('#finish-region').click(function () {
     finishRegion();
     });
 
-$('#submit-gid').click(function() {
-       $.ajax({
-            url: 'get-json-data',
-            data: {gid : $('#area-of-study').val()},
-            dataType: 'json', 
-            success: function(data) {
-                $('#json-data').html('<pre>' + JSON.stringify(data) + '<pre>');
-            },
-            error: function(xhr, status, err) {
-                console.error('graphs', status, err.toString());
+$('#save-graph').click(function () {
+    $.ajax({
+        url: 'insert-graph',
```

Comment:
That's not a bad thought, but the way this is set up now, it is possible for a graph to be inserted while its corresponding elements are not. E.g., if the JSON data is not parsed properly, which actually also happened to me while I was testing this.

It will be possible in the future to check for duplicate names through some other method.